backend/src/main/java/fk/prof/backend/http/LeaderHttpVerticle.java

```diff
@@ -255,19 +250,20 @@ private void handleCreatePolicy(RoutingContext context) {
 
   private void handleUpdatePolicy(RoutingContext context) {
     try {
-      Recorder.ProcessGroup pG = parseProcessGroup(context);
-      PolicyDTO.VersionedPolicyDetails versionedPolicyDetails = parseVersionedPolicyFromPayload(context);
-      policyStore.updateVersionedPolicy(pG, versionedPolicyDetails).setHandler(ar -> setResponse(ar, context));
+      Recorder.ProcessGroup pg = parseProcessGroup(context);
+      PolicyDTO.VersionedPolicyDetails versionedPolicyDetails = ProtoUtil.buildProtoFromBuffer(PolicyDTO.VersionedPolicyDetails.parser(), context.getBody());
```

Comment:
encapsulate this and validation below in `parseVersionedPolicyFromPayload` method